### PR TITLE
[flutter_tools] fix path escaping on in depfile generation

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

When a depfile is generated for gradle build, empty spaces ` ` need to be escaped when creating the depfile. I verified this works locally on my computer, and created a unit test for the right escaping.

https://github.com/flutter/flutter/issues/16604

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_system/depfile.dart [hunk 1]
```diff
@@ -75,7 +75,9 @@ class Depfile {
     for (final File outputFile in files) {
       if (globals.platform.isWindows) {
         // Paths in a depfile have to be escaped on windows.
-        final String escapedPath = outputFile.path.replaceAll(r'\', r'\\');
+        final String escapedPath = outputFile.path
+          .replaceAll(r'\', r'\\')
+          .replaceAll(r' ', r'\ ');
         buffer.write(' $escapedPath');
       } else {
         buffer.write(' ${outputFile.path}');
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_system/depfile.dart]: Does `_separatorExpr` imply that spaces have to be escaped on all platforms? That is, a space preceded by anything but a '\' is a separator not just on Windows, but on other platforms as well.

File: packages/flutter_tools/test/general.shard/build_system/depfile_test.dart [hunk 2]
```diff
@@ -66,6 +67,23 @@ C:\\a.txt: C:\\b.txt
...
+      ..createSync();
+    final Depfile depfile = Depfile(<File>[inputFile], <File>[outputFile]);
+    final File outputDepfile = globals.fs.file('depfile');
+    depfile.writeToFile(outputDepfile);
+
+    expect(outputDepfile.readAsStringSync(), contains(r'C:\\Hello\ Flutter\\a.txt'));
+    expect(outputDepfile.readAsStringSync(), contains(r'C:\\Hello\ Flutter\\b.txt'));
+  }, overrides: <Type, Generator>{
+    FileSystem: () => MemoryFileSystem(style: FileSystemStyle.windows),
+    Platform: () => FakePlatform(operatingSystem: 'windows'),
```
Comment (zanderso) [inline packages/flutter_tools/test/general.shard/build_system/depfile_test.dart]: Test spaces in paths on other platforms, too.

**Comment by jonahwilliams** [comment]
I double checked, and you're right - they have to be escaped on macOS/Linux too. Updating PR. 🤦‍♂

File: packages/flutter_tools/test/general.shard/build_system/depfile_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/build_system/depfile_test.dart]: Done

**Review by zanderso — APPROVED** [review]

**Pull request merged**